core/core-impl/src/test/java/ai/timefold/solver/core/api/solver/SolverFactoryTest.java

```diff
@@ -128,10 +137,24 @@ void createFromInvalidXmlFile_failsShowingBothPathAndReason() throws IOException
 
     @Test
     void create() {
-        SolverConfig solverConfig = PlannerTestUtils.buildSolverConfig(TestdataSolution.class, TestdataEntity.class);
-        SolverFactory<TestdataSolution> solverFactory = SolverFactory.create(solverConfig);
-        Solver<TestdataSolution> solver = solverFactory.buildSolver();
-        assertThat(solver).isNotNull();
+        { // No config override
+            SolverConfig solverConfig = PlannerTestUtils.buildSolverConfig(TestdataSolution.class, TestdataEntity.class);
+            SolverFactory<TestdataSolution> solverFactory = SolverFactory.create(solverConfig);
+            Solver<TestdataSolution> solver = solverFactory.buildSolver();
+            assertThat(solver).isNotNull();
+        }
+
+        { // Config override
+            SolverConfig solverConfig = PlannerTestUtils.buildSolverConfig(TestdataSolution.class, TestdataEntity.class);
+            SolverFactory<TestdataSolution> solverFactory = SolverFactory.create(solverConfig);
+            SolverConfigOverride configOverride = mock(SolverConfigOverride.class);
+            TerminationConfig terminationConfig = new TerminationConfig();
+            terminationConfig.withSpentLimit(Duration.ofSeconds(60));
+            doReturn(terminationConfig).when(configOverride).getTerminationConfig();
+            Solver<TestdataSolution> solver = solverFactory.buildSolver(configOverride);
```

Comment:
Arguably, we also want to support the override directly on `solver.solve(...)`. But let's see how hard that would be; maybe we decide not to do that in the end.